### ISSUE TITLE
How can I replace first two bytes in inputstream?

### ISSUE THREAD

**Issue by feelfreelinux** [brianwernick/ExoMedia#576]
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests

###### Include the following:
 - ExoMedia version: `4.1.0`
 - Device OS version: `8.1`
 - Devide Manufacturer: `Google`
 - Device Name: `Pixel XL`
 
###### Reproduction Steps
I am trying to implement coub.com player in my app. However, in order to decode my video as .mp4 I need to replace two first bytes with 0x00. https://github.com/rg3/youtube-dl/issues/13754#issuecomment-319839806
I tried to create custom DataSource that does it, but it looks like ExoMedia does not recognize it correctly, and both video and seekbar stops after few seconds. Is it possible to achieve it using this library? Should I write custom MediaSource?

**Comment by brianwernick** [comment]
You'll need to provide a custom http client implementation (see #86) that handles zeroing out those bits.

**Comment by feelfreelinux** [comment]
@brianwernick I created a custom OkHttpDataSource implementation, set it in ExoMedia as default and it plays correctly. However, LoopingMediaSource does not loop this video, and seekbar shows that this video has only 1 second. I checked, and coub.com does not supply Content-Length in headers, but puts it in API response. I tried to override contentlength in my custom OkHttpDataSource, but its still broken.
